presto-hive/src/main/java/com/facebook/presto/hive/InternalHiveSplit.java

```diff
@@ -213,19 +194,20 @@ public int getEstimatedSizeInBytes()
     {
         int result = INSTANCE_SIZE;
         result += path.length() * Character.BYTES;
-        result += sizeOfObjectArray(partitionKeys.size());
-        for (HivePartitionKey partitionKey : partitionKeys) {
-            result += partitionKey.getEstimatedSizeInBytes();
-        }
         result += sizeOfObjectArray(blocks.size());
         for (InternalHiveBlock block : blocks) {
             result += block.getEstimatedSizeInBytes();
         }
-        result += partitionName.length() * Character.BYTES;
-        result += sizeOfObjectArray(columnCoercions.size());
-        for (HiveTypeName hiveTypeName : columnCoercions.values()) {
-            result += INTEGER_INSTANCE_SIZE + hiveTypeName.getEstimatedSizeInBytes();
+
+        // partitionInfo is shared between all InternalHiveSplits for a partition
```

Comment:
Maybe also add Java doc for `HiveSplitPartitionInfo` class as well so the community can understand the purpose of these classes :). What you wrote here is already quite good, maybe also add some thing like "so each partition is only accounted once for memory tracking"